pkg/webhook/nodenetworkconfigurationpolicy/conditions_test.go

```diff
@@ -0,0 +1,127 @@
+package nodenetworkconfigurationpolicy
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var _ = Describe("Mutating Admission Webhook", func() {
+	type WebhookCase struct {
+		Policy nmstatev1alpha1.NodeNetworkConfigurationPolicy
+	}
+	DescribeTable("when reset condition is called", func(c WebhookCase) {
+		var (
+			testStartTime    int64
+			request          = webhook.AdmissionRequest{}
+			obtainedResponse webhook.AdmissionResponse
+			patchedPolicy    = nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
+		)
+		s := scheme.Scheme
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion,
+			&nmstatev1alpha1.NodeNetworkConfigurationPolicy{},
+		)
+		c.Policy.Status.Conditions.Set(
+			nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
+			corev1.ConditionTrue,
+			nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+			"")
+
+		cli := fake.NewFakeClientWithScheme(s, &c.Policy)
+		testStartTime = time.Now().UnixNano()
+
+		By("invoking the webhook")
+		data, err := json.Marshal(c.Policy)
+		Expect(err).ToNot(HaveOccurred())
+		request.Object = runtime.RawExtension{
+			Raw: data,
+		}
+		obtainedResponse = resetConditionsHook().Handle(context.TODO(), request)
+
+		By("patching the policy with the result")
+		patch := client.ConstantPatch(types.JSONPatchType, obtainedResponse.Patch)
+		policy := c.Policy.DeepCopy()
+		err = cli.Patch(context.TODO(), policy, patch)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Retrieve the patched policy")
+		err = cli.Get(context.TODO(), types.NamespacedName{Name: ""}, &patchedPolicy)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(obtainedResponse.Allowed).To(BeTrue(), "Mutation not allowed")
```

Comment:
This whole test is quite a wild west. It would help if there was something I could read to get some hints about the code. Could we change the explanation string to say "Mutation of the request should be allowed". Similar in the following checks "Status reason should match the expected". "There should be exactly two patches in the response"